### ISSUE TITLE
data verify conditionals on startup 

### ISSUE THREAD

**Issue by ThePotatoGuy** [Monika-After-Story/MonikaModDev#5683]
if an event has a conditional that crashes when ran, we should log it and not hard crash.
this may occur on downgrades and other unsupported actions, but its worth having some resiliency.

### key notes
* conditional checking must happen at **runtime** because of lack of available functions
* we should log any instances of a exception occuring when a conditional runs
* if an event's conditional crashes in this check, we should clear the event data completely.

**Comment by multimokia** [comment]
> if an event's conditional crashes in this check, we should clear the event data completely.

Not sure about this, because then we lose actual event data which *is* valid. If anything, only conditional, start_date, end_date, and action should be cleared.

(That or clear it after copying the important data, and just transfer the rest of the bits over)

**Comment by ThePotatoGuy** [comment]
clearing only conditional and related data leaves the event in a potential dummy state where it's current status (random/pool/unlocked/locked/etc...) is unpredictable. Clearing the event means it gets reset to the default state on next launch which is predictable.

if we really want to preserve data then I could add a flag or or something to mark the event as tainted so only some of its values are reset upon next launch

**Comment by ThePotatoGuy** [comment]
@Booplicate we check conditionals on start now right? since you added precompiling to conditionals?

**Comment by Booplicate** [comment]
Indeed, I moved it to `splashscreen` after affection system init, so this
> conditional checking must happen at runtime because of lack of available functions

Should be solved now.


Also, I think we should still raise in dev mode, while for users it should just log. Although, I'm not sure about resetting ev props. As you said, if somebody downgrades, ev breaks because of the `conditional`. If you clear the ev data, the user won't be able to get the event even after they install the proper version where it's fixed. Because now the ev has no trigger points or they are invalid.


> if we really want to preserve data then I could add a flag or or something to mark the event as tainted so only some of its values are reset upon next launch

So I'd say this is a better solution. But then maybe we could make it so the ev doesn't exist at all whiel it's in the broken state?

**Comment by ThePotatoGuy** [comment]
>So I'd say this is a better solution. But then maybe we could make it so the ev doesn't exist at all whiel it's in the broken state?

I like this
